bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

```diff
@@ -335,16 +335,21 @@ public Capabilities getCapabilities(@Nullable Integer capabilitiesId) {
     /**
      * Return a Capabilities class instance that corresponds to the given 'typeId' parameter. If the 'typeId' parameter
      * is a valid type in the database, and it has a 'capabilitiesOverride' value, then an instance of the respective
-     * overridden Capabilities class is returned. Otherwise if the 'capabilitiesId' parameter is for a valid
-     * capabilities entry in the database, then that respective Capabilities class instance is returned. Otherwise a
-     * blank Capabilities class instance is returned.
+     * overridden Capabilities class is returned. Otherwise if the type is a valid 'capabilities' instance then that is
+     * returned. Otherwise if the 'capabilitiesId' parameter is for a valid capabilities entry in the database, then
+     * that respective Capabilities class instance is returned. Otherwise a blank Capabilities class instance is
+     * returned.
      *
      * @param typeId the target shade type Id (to check if it has a 'capabilitiesOverride' value).
      * @param capabilitiesId the target capabilities value (when type Id does not have a 'capabilitiesOverride').
      * @return corresponding Capabilities class instance.
      */
     public Capabilities getCapabilities(int typeId, @Nullable Integer capabilitiesId) {
-        int targetCapabilities = TYPE_DATABASE.getOrDefault(typeId, new Type()).getCapabilitiesOverride();
+        Type type = TYPE_DATABASE.getOrDefault(typeId, new Type());
+        int targetCapabilities = type.getCapabilitiesOverride();
+        if (targetCapabilities < 0) {
+            targetCapabilities = type.getCapabilities();
```

Comment:
Is it intended that capabilities derived from type should take precedence over capabilities from response? It seems that order of precedence is now:
- Capabilities override from type.
- Capabilities from type.
- Capabilities from response.
- Default capabilities.

I would have expected:
- Capabilities override from type.
- Capabilities from response.
- Capabilities from type.
- Default capabilities.

Otherwise there are actually two levels of override from type, and capabilities from response is not actually used?

Reply:
> Is it intended that capabilities derived from type should take precedence over capabilities from response?

That is really a philosophical question. If the database is correct, then both approaches give the same result. We are also logging if the two capabilities do not match, so we can update the database. I don't have any objection to reversing the logic, so please let me know what do YOU prefer?

Reply:
PS until now, except for the hub1 case, we have not seen any conflicts between the two approaches..

Reply:
IMHO capabilities from the response should be used when present, i.e. take precedence in case of mismatch between database content and actual capabilities. In a way it's a matter of trust - do we trust our database more than data provided by the hub. But also implementation-wise it seems confusing that type will **always override** capabilities (type->capabilities**Override** and then type->capabilities), so that capabilities from the hub is only used in the case where type is unknown.

Perhaps small difference... I'll not insist, although two overrides, where only one of them is called override by name, is kind of tickling my brain. It would also preserve existing behavior, as I assume PR was only intended to fix missing capabilities issue with v1 hub, not change precedence behavior in general?

Reply:
Looking to the future, I suppose it is likely that HD introduces new types within the existing capabilities framework. But even in this case the scenario "type before capabilites" and the scenario "capabilities before type" will still produce the same result.

But let me sleep on this. I will get back to you tomorrow..

Reply:
Ok. I will do it as you suggest. I think also if no capabilities is found then the default value should not be -1 (unknown), but rather 0 (which means that at least some basic functionality will always exist..)